data/model/oci/tag.py

```diff
@@ -363,6 +363,41 @@ def create_temporary_tag_if_necessary(manifest, expiration_sec):
         )
 
 
+def create_temporary_tag_outside_timemachine(manifest):
+    """
+    Creates a temporary tag that is outside the time machine window.
+    Tag is immediately available for garbage collection.
+    """
+    tag_name = "$temp-%s" % str(uuid.uuid4())
+    now_ms = get_epoch_timestamp_ms()
+    try:
+        namespace = (
+            User.select(User.removed_tag_expiration_s)
+            .join(Repository, on=(Repository.namespace_user == User.id))
+            .where(Repository.id == manifest.repository_id)
+            .get()
+        )
+    except User.DoesNotExist:
+        # Should never happen, but throw error anyway
+        raise DataModelException(
+            "Could not find namespace for repository: %s" % manifest.repository_id
+        )
+
+    time_machine_ms = namespace.removed_tag_expiration_s * 1000
+    end_ms = now_ms - time_machine_ms
```

Comment:
Can we use just `0`?

Reply:
Do you mean can we just set the end_ms to 0?

Reply:
yes, can we? Or maybe `1`, or any other well-known timestamp in the past. I assume it'll save us 1 query to the database.

Reply:
Yes we can actually. I typically do this because of the tag name, repository id, lifetime_end_ms unique index on the tag table. Since we can have expired tags that have the same name there is sometimes a clash if we try setting the lifetime_end_ms to the same value, in this case 0 or 1. Since we're using a 4 character hash for the name it shouldn't be an issue. I'll update.

Reply:
It would only be an issue if the manifest was blocked by quota, also while during the push of each blob the quota was not exceeded, and the same 4 character hash was generated